web/src/ui/components/RequestBuilder.js

```diff
@@ -0,0 +1,114 @@
+    const [body, setBody] = useState({})
+    const [bodyDisabled, setBodyDisabled] = useState(true)
+    var headerInputTimeout, bodyInputTimeout
+
+    const handleMethodChange = (selectedMethod) => {
+        setMethod(selectedMethod)
+        enableOptions(selectedMethod)
+    }
+
+    const removeCurrentOptions = () => {
+        setBodyDisabled(true)
+        //...
+    }
+
+    const enableOptions = (selectedMethod) => {
+        removeCurrentOptions()
+        if (selectedMethod.toLowerCase() === "post") {
+            setBodyDisabled(false)
+        }
+        //...
+    }
+
+    const handlePathChange = (event) => {
+        setPath(event.target.value)
+    }
+
+    const handleHeaderChange = (header) => {
+        clearTimeout(headerInputTimeout)
+        headerInputTimeout = setTimeout(() => {
+            if ((header = isValidJson(header))) {
+                setHeader(header)
+            } else {
+                //evtl Error bars triggern etc
+            }
+        }, 2000)
+    }
+    const handleBodyChange = (body) => {
+        clearTimeout(bodyInputTimeout)
+        bodyInputTimeout = setTimeout(() => {
+            if ((body = isValidJson(body))) {
+                setBody(body)
+            } else {
+                //evtl Error bars triggern etc
+            }
+        }, 2000)
+    }
+
+    const handleSubmit = () => {
+        const requestParams = {
+            method,
+            path,
+            header,
+            body
+        }
+        console.log("Submitting request:", requestParams)
+        // Hier kannst du deine Anfrage senden, z.B. mit fetch oder axios
+    }
+
+    const selectProps = {
+        options: ["POST", "GET", "DELETE", "PUT", "HEAD"],
+        label: "method",
+        className:
+            "bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500",
+        onSelect: handleMethodChange
+    }
+    return (
+        <div className="w-1/2 h-full bg-gray-800 p-4">
+            <div className="text-white text-2xl">Request Builder</div>
+            {/* Method selection */}
+            <Select {...selectProps} />
+            {/* Path */}
+            <input
+                type="text"
+                value={path}
+                onChange={handlePathChange}
+                placeholder="Path"
+                className="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 mt-2"
+            />
+            {/* Header */}
+            <JsonTextarea placeholder="Header" onInput={handleHeaderChange} />
+            {/* Body */}
+            <JsonTextarea
```

Comment:
Um das noch ein bisschen zu erweitern: die JSON-Eingabe wird ja nur benötigt, wenn auch der Content-Type auf JSON steht, ansonsten habe wir eine einfache Text-Area ohne Validierung. Wir können hier noch einen Umschalter über der Area machen mit "Raw" / "JSON", welcher zwischen den TextAreas umschaltet und beim Klick auf Raw bzw. JSON auch den Content-Type entsprechend im Header überschreibt. Defaultmäßig könnten wir den JSON aktiv haben.

Reply:
Wäre es dann von Überlegung, eine Komponente zu haben, welche Allgemein Textarea heißt, die aber mit einem Prop contentType dann entsprechend anders agiert?

Reply:
ContentType ist ja eigentlich ein Header-Property von dem eine Textarea nicht weiß - würde die Text-Area-Komponente immer nur ein HTML-Text-Area sein lassen mit setter und value. Unsere JsonTextArea ist dann eine Komponente welche die normale Text-Area-Komponente verwendet und sich einfach in den setter hängt um noch eine Validierung (und evt. auch den Content-Type change) zu machen und evt. unter der Text-Area noch ein div setzt, wo die Info angezeigt. wird.